From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 24 Mar 2014 23:17:25 +0000
Subject: firewaire: Avoid ABI change in 3.13.7

struct fw_device gained a new member.  This is used by all firewire
drivers but is only allocated by firewire-core and the new field is
also reserved to firewire-core.  So move it to the end and hide it
from genksyms.

--- a/include/linux/firewire.h
+++ b/include/linux/firewire.h
@@ -200,9 +200,11 @@ struct fw_device {
 	unsigned irmc:1;
 	unsigned bc_implemented:2;
 
-	work_func_t workfn;
 	struct delayed_work work;
 	struct fw_attribute_group attribute_group;
+#ifndef __GENKSYMS__
+	work_func_t workfn;
+#endif
 };
 
 static inline struct fw_device *fw_device(struct device *dev)
